### ISSUE TITLE
  UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory. This may consume a large amount of memory.

### ISSUE THREAD

**Issue by louisly** [eragonruan/text-detection-ctpn#27]
when i try to train my own datasets, i faced this core dump,

Computing bounding-box regression targets...
bbox target means:
[[ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]]
[ 0.  0.  0.  0.]
bbox target stdevs:
[[ 0.1  0.1  0.2  0.2]
 [ 0.1  0.1  0.2  0.2]]
[ 0.1  0.1  0.2  0.2]
Normalizing targets
done
Solving...
/data/resys/var/python2.7.3/lib/python2.7/site-packages/tensorflow/python/ops/gradients_impl.py:91: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
Segmentation fault (core dumped)

 this fault caused by the function train_model()   in lib/fast-rcnn/train.py ，  when it run train_op=opt.apply_gradients(list(zip(grads,tvars),global_step=global_step)) .

have you ever faced this error？

**Comment by eragonruan** [comment]
@louisly you can check this [link](https://github.com/tensorflow/fold/issues/13)

**Comment by louisly** [comment]
@eragonruan thanks!  i use the virtualenv solve this problem.
but i faced a new error:

assign pretrain model biases to conv1_2
assign pretrain model weights to conv2_2
assign pretrain model biases to conv2_2
assign pretrain model weights to conv2_1
assign pretrain model biases to conv2_1
E tensorflow/stream_executor/cuda/cuda_blas.cc:372] failed to create cublas handle: CUBLAS_STATUS_NOT_INITIALIZED
W tensorflow/stream_executor/stream.cc:1390] attempting to perform BLAS operation using StreamExecutor without BLAS support

i guess it may be the difference between tensorflow1.0 and tensorflow 1.2,
which tensorflow have you used? tensorflow 1.2.0?

**Comment by eragonruan** [comment]
@louisly I am using tf1.3, This seems bug about tf and opencv, you can try update your tf version. see this [link](https://stackoverflow.com/questions/42488615/failed-to-create-cublas-handle-tensorflow-interaction-with-opencv)

**Comment by louisly** [comment]
@eragonruan  i solved the CUBLAS_STATUS_NOT_INITIALIZED error，just add the gpu memory.
i have 4 gpu, and when  i use 3 of them that the error disappeared, it cost too much memory of gpu.  but when i use 1 or 2  gpu, it can not work.  
is there anything wrong? how much memory of gpu have you used?

**Comment by eragonruan** [comment]
@louisly I have two GTX1070, 8G each. for me, 1 or 2 gpu is ok. this problem may related to your environment.

**Comment by louisly** [comment]
@eragonruan  thanks!    i used almost 7000 images to train, i don't know whether the num of images matters. 
and can you tell me  the total loss , rpn_loss_cls, rpn_loss_box  of you model after 50000 iters,  i just want to know the loss score which may lead a good performance to decide when to stop my training

**Comment by eragonruan** [comment]
@louisly about 0.6， 0.08， 0.25

**Comment by zhangsn828** [comment]
@louisly How do you solve this problem?
